app/client/components/GridView.js

```diff
@@ -908,14 +908,22 @@ GridView.prototype.makeHeadersFromRow = async function(selection) {
   if (this._getCellContextMenuOptions().disableMakeHeadersFromRow){
     return;
   }
+  const record = this.tableModel.tableData.getRecord(selection.rowIds[0]);
+  const formatter = selection.fields[0].formatter();
```

Comment:
Are you sure the formatter will always be the same whatever the field is?

Reply:
All Formatters extends `BaseFormatter` that have the public function `formatAny`, so don't matter if there are different formatters 
Can see it in `ValueFormatter.ts` file

Reply:
`formatAny` calls `format`, which _can_ be overridden by subclasses of `BaseFormatter`. For example, fields for Numeric columns use `NumericFormatter.format` to format numbers according to options configured in the creator panel.

One thing to check would be what happens when you try to format a reference column that has a value, but the referenced value is null (i.e. a "blank" reference). My memory is that it returns an empty string, and if that's indeed the case, then we should be careful not to try to set column labels to blank strings.

Reply:
Making more test, it's not a good thing using this field formatter, because it's the formatter of the field selected by the user and not the formatter of the specfic cell we are processing in the row. 
I will see if we can get a field from the colId and the rowId in a tableData